apps/src/music/views/MusicView.jsx

```diff
@@ -616,48 +619,70 @@ class UnconnectedMusicView extends React.Component {
+        id="instructions-area"
         className={classNames(
           moduleStyles.instructionsArea,
           position === InstructionsPositions.TOP
             ? moduleStyles.instructionsTop
-            : moduleStyles.instructionsSide,
-          position === InstructionsPositions.LEFT &&
-            moduleStyles.instructionsLeft,
-          position === InstructionsPositions.RIGHT &&
-            moduleStyles.instructionsRight
+            : moduleStyles.instructionsSide
         )}
       >
-        <Instructions
-          progressionStep={this.progressManager.getProgressionStep()}
-          currentLevelIndex={this.props.currentLevelIndex}
-          levelCount={this.props.levelCount}
-          onNextPanel={this.onNextPanel}
-          baseUrl={baseUrl}
-          vertical={position !== InstructionsPositions.TOP}
-          right={position === InstructionsPositions.RIGHT}
-        />
+        <PanelContainer
+          id="instructions-panel"
+          headerText={musicI18n.panelHeaderInstructions()}
+        >
+          <Instructions
+            progressionStep={this.progressManager.getProgressionStep()}
+            showProgressionStep={!this.hasLevels()}
+            currentLevelIndex={this.props.currentLevelIndex}
+            levelCount={this.props.levelCount}
+            onNextPanel={this.onNextPanel}
+            baseUrl={baseUrl}
+            vertical={position !== InstructionsPositions.TOP}
+            right={position === InstructionsPositions.RIGHT}
+          />
+        </PanelContainer>
       </div>
     );
   }
 
-  renderTimelineArea(timelineAtTop, instructionsOnRight) {
+  renderPlayArea(timelineAtTop) {
     return (
       <div
-        id="timeline-area"
+        id="play-area"
         className={classNames(
-          moduleStyles.timelineArea,
-          timelineAtTop ? moduleStyles.timelineTop : moduleStyles.timelineBottom
+          moduleStyles.playArea,
+          timelineAtTop ? moduleStyles.playAreaTop : moduleStyles.playAreaBottom
         )}
       >
-        <Controls
-          setPlaying={this.setPlaying}
-          playTrigger={this.playTrigger}
-          top={timelineAtTop}
-          instructionsAvailable={!!this.progressManager}
-          toggleInstructions={() => this.toggleInstructions(false)}
-          instructionsOnRight={instructionsOnRight}
-        />
-        <Timeline />
+        <div id="controls-area" className={moduleStyles.controlsArea}>
+          <PanelContainer
+            id="controls-panel"
+            headerText={musicI18n.panelHeaderControls()}
+          >
+            <Controls
+              setPlaying={this.setPlaying}
+              playTrigger={this.playTrigger}
+              top={this.props.timelineAtTop}
+              instructionsAvailable={!!this.progressManager}
+              toggleInstructions={() => this.toggleInstructions(false)}
+              instructionsOnRight={false}
+              hasTrigger={this.musicBlocklyWorkspace.hasTrigger.bind(
```

Comment:
As an alternative to plumbing `hasTrigger` through these components, maybe we could save the trigger IDs that are currently used in the project in the redux store? This would just need to get dispatched from MusicView whenever code has changed.